src/Uno.UI/UI/Xaml/Media/Imaging/BitmapSource.cs

```diff
@@ -16,16 +16,17 @@ public partial class BitmapSource : ImageSource
 		public int PixelHeight
 		{
 			get { return (int)GetValue(PixelHeightProperty); }
-			internal set { SetValue(PixelHeightProperty, value); }
+#if __WASM__ // The setter should be private protected and must not be set by Image. In fact, the PixelWidth/PixelHeight value should be available even without being rendered by Image.
+			internal
+#else
+			private protected
+#endif
```

Comment:
Note to self that I don't forget:

This comment is incorrect. `Image` should update PixelWidth/PixelHeight when BitmapImage is constructed via a Uri.

This should be done before raising BitmapImage.ImageOpened here:

https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Controls/Image/Image.cs#L100

https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.skia.cs#L92


https://github.com/unoplatform/uno/blob/27c7e130a6a558a746639041a32b468fe26d469b/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs#L189